illumos-utils/src/opte/firewall_rules.rs

```diff
@@ -61,41 +61,39 @@ impl FromVpcFirewallRule for VpcFirewallRule {
     }
 
     fn hosts(&self) -> Vec<Address> {
-        self.filter_hosts.as_ref().map_or_else(
-            || vec![Address::Any],
-            |hosts| {
-                hosts
-                    .iter()
-                    .map(|host| match host {
-                        HostIdentifier::Ip(IpNet::V4(net))
-                            if net.prefix() == 32 =>
-                        {
-                            Address::Ip(IpAddr::Ip4(net.ip().into()))
-                        }
-                        HostIdentifier::Ip(IpNet::V4(net)) => {
-                            Address::Subnet(IpCidr::Ip4(Ipv4Cidr::new(
-                                net.ip().into(),
-                                Ipv4PrefixLen::new(net.prefix()).unwrap(),
-                            )))
-                        }
-                        HostIdentifier::Ip(IpNet::V6(net))
-                            if net.prefix() == 128 =>
-                        {
-                            Address::Ip(IpAddr::Ip6(net.ip().into()))
-                        }
-                        HostIdentifier::Ip(IpNet::V6(net)) => {
-                            Address::Subnet(IpCidr::Ip6(Ipv6Cidr::new(
-                                net.ip().into(),
-                                Ipv6PrefixLen::new(net.prefix()).unwrap(),
-                            )))
-                        }
-                        HostIdentifier::Vpc(vni) => {
-                            Address::Vni(Vni::new(u32::from(*vni)).unwrap())
-                        }
-                    })
-                    .collect::<Vec<Address>>()
-            },
-        )
+        match self.filter_hosts {
+            Some(ref hosts) if hosts.len() > 0 => hosts
```

Comment:
The copy of clippy running in my brain saw this and suggested `!hosts.is_empty()`.